components/builder-api/src/config.rs

```diff
@@ -210,6 +212,28 @@ impl ToSocketAddrs for HttpCfg {
     }
 }
 
+#[derive(Clone, Debug, Deserialize)]
+#[serde(default)]
+pub struct KafkaCfg {
+    pub bootstrap_nodes:     String, // "foo.bar.baz:9090,baz.fuzz.bizz:9191"
```

Comment:
Maybe `Vec<String>` if it's possible to have more than one?

Reply:
So this is something I hate in this implementation. However rdkafka-rust expects the bootstrap list as a single node. Which would mean unnecessarily doing some string parsing to put this into the shape of Vec<String> only to then reverse that work to get it back into this shape.

Reply:
The way I look at things like this is that we have our API (the thing we use every day, and that is part of our core business logic), and then we have the API of whatever libraries we happen to be using today. Ideally, our API would be as logical as we could make it; putting aside Kafka for a second, if any of us were creating any kind of data structure that called for more than one of something, I'm pretty sure we'd all reach for something like a `Vec` first. Inside of that API, we'd do whatever manipulations are necessary to interact with the libraries we're using. Today, we're using `rdkafka-rust`, which wants everything as a comma-delimited String. Tomorrow we might use something completely different that expects some other format (maybe something weird like a semicolon-delimited string :shrug: ). The take away here is that our business logic doesn't have to change if we keep a `Vec`, and you only have to tweak the translation layer between that API and the library, rather than having to tweak _everything_.

(As an aside, it's always nicer to encode logic and expectations as code, rather than comments.)

I readily acknowledge that it's super small stakes right now, but little decisions early on have a tendency to cause all kinds of issues down the road.

Reply:
100% agreement.